app/models/name.rb

```diff
@@ -408,33 +435,95 @@ class Name < AbstractModel
     end
   end
 
-  scope :with_rank,
-        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  include ScopesForTimestamps
 
+  scope :of_lichens, -> { where(Name[:lifeform].matches("%lichen%")) }
+  scope :not_lichens, -> { where(Name[:lifeform].does_not_match("% lichen %")) }
+  scope :deprecated, -> { where(deprecated: true) }
   scope :not_deprecated, -> { where(deprecated: false) }
-
+  scope :with_description, -> { where.not(description_id: nil) }
+  scope :without_description, -> { where(description_id: nil) }
   ### Module Name::Spelling
   scope :with_correct_spelling, -> { where(correct_spelling_id: nil) }
-
-  # For a glitch discovered in the wild:
+  scope :with_incorrect_spelling, -> { where.not(correct_spelling_id: nil) }
   scope :with_self_referential_misspelling, lambda {
     where(Name[:correct_spelling_id].eq(Name[:id]))
   }
-
+  scope :with_synonyms, -> { where.not(synonym_id: nil) }
+  scope :without_synonyms, -> { where(synonym_id: nil) }
   ### Module Name::Taxonomy
-  scope :with_classification_like,
-        # Use multi-line lambda literal because fixtures blow up with "lambda":
-        # NoMethodError: undefined method `ranks'
-        #   test/fixtures/names.yml:28:in `get_binding'
-        ->(rank, text_name) { # rubocop:disable Style/Lambda
-          where(Name[:classification].matches("%#{rank}: _#{text_name}_%"))
-        }
-
-  scope :with_name_like,
-        ->(text_name) { where(Name[:text_name].matches("#{text_name} %")) }
-
+  scope :with_rank,
+        ->(rank) { where(rank: Name.ranks[rank]) if rank }
   scope :with_rank_below,
         ->(rank) { where(Name[:rank] < Name.ranks[rank]) }
```

Comment:
Is the conditional also needed here, as in `scope :with_rank`, like this:
```suggestion
        ->(rank) { where(Name[:rank] < Name.ranks[rank]) if rank }
```

Reply:
Hm. I think VS Code autocorrect added that `if rank`, so I don't know! Maybe it evaluates to `nil < nil` if there's no rank?

Reply:
```ruby
3.0.4 :009 > Name.first.rank < nil
  Name Load (0.3ms)  SELECT `names`.* FROM `names` ORDER BY `names`.`id` ASC LIMIT 1
(irb):9:in `<': comparison of String with nil failed (ArgumentError)                   
```

Reply:
QED!

Reply:
Committed!